### PR TITLE
feat: add missing warning when action is skipped 

### LINKED ISSUE #436: Add more logging
As discussed in https://github.com/fastify/github-action-merge-dependabot/issues/431#issuecomment-1574126076

> To the extent it could print [details about why something didn't run, such as commit verification failing], it would make it much easier to debug...

### PR DESCRIPTION
**Pull request by guilhermelimak**
This PR adds a warning message when the actor verification fails and the action execution is skipped. Previously the action would just not execute and no messages would be logged.

Resolves #436.

### CODE REVIEW

File: action.yml
Comment (simoneb) [inline action.yml:62]: rather than this solution, shall we not just keep (add) checks in the JS code and keep all the warnings there?

Comment (guilhermelimak) [reply action.yml:62]: I think it can be removed

Comment (guilhermelimak) [reply action.yml:62]: I'm not entirely sure why we did that in the first place but I think it should be fine to remove it and leave it up to the action.

Comment (guilhermelimak) [reply action.yml]: I've just removed and it all seems to work fine when testing. I also looked it up and [this](https://github.com/fastify/github-action-merge-dependabot/pull/276/files/3ae29fea7e07144acf343022ab10ee66ce643198#r975431852) was when it was added so I don't think it had a very strong reason, just not running the action at all if not needed to maybe save resources or something like that

File: action.yml
Comment (simoneb) [inline action.yml:50]: I'm not sure about the removal of this, since we don't control this action there's a chance that the action fails if executed without this condition. to be sage, I would just remove the condition from the execution of our own action, but also make sure that the same logic is applied within the action js code.

as for why this may have been added, hard to say, but I guess to short-circuit and avoid execution if unnecessary.

ideally I'd like to remove the yaml condition in both actions, but I fear that this action we don't control will break

Comment (guilhermelimak) [reply action.yml]: yep, that makes sense.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
What is the benefit?

**Comment by simoneb** [comment]
> What is the benefit?

Print warning messages when the action doesn't get executed. The conditions in the yaml prevented the execution altogether, including printing warning messages.

**Review by simoneb — APPROVED** [review]

File: action.yml
Comment (Eomm) [inline action.yml:56]: I would remove `inputs.skip-verification == 'true'` only - should it work?

Comment (guilhermelimak) [reply action.yml:56]: @Eomm I don't think so, from my tests it's the `github.actor == 'dependabot[bot]'` check that's failing

Comment (simoneb) [reply action.yml:56]: ok folks so here's the rationale, since there seem to be some questions in this regard:

- we want to display warnings when the action's execution is skipped
- if we skip the action execution in the yaml, we don't have an easy way to display warnings (unless we put the warnings in the yaml, which was the first solution @guilhermelimak went for and which I don't like much)
- the action is using an external action https://github.com/dependabot/fetch-metadata/, which IIRC assumes that the action is built run in a dependabot context
- so what we want to do, is keep skipping that action without thereby changing the conditions for its executions
- remove the conditions of execution from the invocation of our custom code so we can handle them in JS code and print warnings accordingly

**Comment by simoneb** [comment]
@Uzlopak waiting for your decision on this PR

**Comment by max-sixty** [comment]
Thanks a lot!

**Pull request merged**
